```
Error running mint automation
+ mkdir -p 10432-bcb6332/gopath/src/github.com/minio
+ git -C 10432-bcb6332/gopath/src/github.com/minio clone --quiet https://github.com/minio/minio.git
+ git -C 10432-bcb6332/gopath/src/github.com/minio/minio remote add minio https://github.com/minio/minio.git
+ git -C 10432-bcb6332/gopath/src/github.com/minio/minio fetch --quiet minio pull/10432/head:pr10432
+ git -C 10432-bcb6332/gopath/src/github.com/minio/minio checkout --quiet pr10432
+ GO111MODULE=on
+ GOPATH=/home/mint-auto/mint-auto/10432-bcb6332/gopath
+ make -C 10432-bcb6332/gopath/src/github.com/minio/minio --quiet
go: downloading github.com/minio/minio-go/v7 v7.0.5-0.20200811211821-14ed05478889
go: downloading gopkg.in/jcmturner/gokrb5.v7 v7.3.0
go: downloading github.com/dustin/go-humanize v1.0.0
go: downloading github.com/minio/highwayhash v1.0.0
go: downloading github.com/gorilla/mux v1.8.0
go: downloading github.com/gorilla/rpc v1.2.0
go: downloading github.com/cespare/xxhash/v2 v2.1.1
go: downloading go.etcd.io/etcd/v3 v3.3.0-rc.0.0.20200707003333-58bb8ae09f8e
go: downloading github.com/Azure/azure-pipeline-go v0.2.2
go: downloading github.com/valyala/tcplisten v0.0.0-20161114210144-ceec8f93295a
go: downloading github.com/willf/bloom v2.0.3+incompatible
go: downloading github.com/dchest/siphash v1.2.1
go: downloading github.com/Azure/azure-storage-blob-go v0.10.0
go: downloading github.com/mattn/go-ieproxy v0.0.1
go: downloading github.com/mattn/go-colorable v0.1.4
go: downloading github.com/minio/sha256-simd v0.1.1
go: downloading github.com/colinmarc/hdfs/v2 v2.1.1
go: downloading github.com/cheggaaa/pb v1.0.28
go: downloading golang.org/x/crypto v0.0.0-20200709230013-948cd5f35899
go: downloading github.com/shirou/gopsutil v2.20.3-0.20200314133625-53cec6b37e6a+incompatible
go: downloading github.com/gorilla/handlers v1.4.2
go: downloading github.com/spaolacci/murmur3 v1.1.0
go: downloading github.com/hashicorp/vault/api v1.0.4
go: downloading github.com/klauspost/compress v1.10.3
go: downloading cloud.google.com/go v0.39.0
go: downloading github.com/dgrijalva/jwt-go v3.2.0+incompatible
go: downloading github.com/montanaflynn/stats v0.5.0
go: downloading github.com/hashicorp/hcl v1.0.0
go: downloading github.com/prometheus/client_golang v1.0.0
go: downloading google.golang.org/api v0.5.0
go: downloading github.com/hashicorp/vault/sdk v0.1.13
go: downloading github.com/prometheus/procfs v0.0.8
go: downloading github.com/ncw/directio v1.0.5
go: downloading github.com/hashicorp/go-retryablehttp v0.5.4
go: downloading github.com/googleapis/gax-go/v2 v2.0.4
go: downloading github.com/mattn/go-isatty v0.0.8
go: downloading github.com/hashicorp/go-multierror v1.0.0
go: downloading github.com/secure-io/sio-go v0.3.0
go: downloading github.com/hashicorp/errwrap v1.0.0
go: downloading golang.org/x/sys v0.0.0-20200625212154-ddb9806d33ae
go: downloading google.golang.org/genproto v0.0.0-20190819201941-24fa4b261c55
go: downloading github.com/beorn7/perks v1.0.0
go: downloading github.com/djherbis/atime v1.0.0
go: downloading github.com/mitchellh/go-homedir v1.1.0
go: downloading google.golang.org/grpc v1.26.0
go: downloading github.com/hashicorp/go-cleanhttp v0.5.1
go: downloading github.com/minio/md5-simd v1.1.0
go: downloading github.com/klauspost/reedsolomon v1.9.9
go: downloading github.com/rs/cors v1.7.0
go: downloading github.com/minio/cli v1.22.0
go: downloading golang.org/x/net v0.0.0-20200707034311-ab3426394381
go: downloading github.com/gogo/protobuf v1.3.1
go: downloading github.com/hashicorp/go-sockaddr v1.0.2
go: downloading gopkg.in/square/go-jose.v2 v2.3.1
go: downloading github.com/tinylib/msgp v1.1.2
go: downloading github.com/klauspost/cpuid v1.3.1
go: downloading github.com/klauspost/readahead v1.3.1
go: downloading git.apache.org/thrift.git v0.13.0
go: downloading golang.org/x/time v0.0.0-20190308202827-9d24e82272b4
go: downloading github.com/mattn/go-runewidth v0.0.2
go: downloading github.com/mitchellh/mapstructure v1.1.2
go: downloading github.com/alecthomas/participle v0.2.1
go: downloading github.com/rs/xid v1.2.1
go: downloading github.com/prometheus/common v0.4.1
go: downloading gopkg.in/olivere/elastic.v5 v5.0.86
go: downloading github.com/nats-io/stan.go v0.7.0
go: downloading github.com/pkg/errors v0.9.1
go: downloading github.com/google/uuid v1.1.1
go: downloading go.uber.org/zap v1.14.1
go: downloading github.com/tidwall/gjson v1.3.5
go: downloading go.opencensus.io v0.22.3
go: downloading golang.org/x/oauth2 v0.0.0-20190226205417-e64efc72b421
go: downloading github.com/Shopify/sarama v1.24.1
go: downloading github.com/miekg/dns v1.1.8
go: downloading github.com/streadway/amqp v0.0.0-20190404075320-75d898a42a94
go: downloading github.com/coredns/coredns v1.4.0
go: downloading github.com/tidwall/pretty v1.0.0
go: downloading github.com/golang/groupcache v0.0.0-20190702054246-869f871628b6
go: downloading github.com/eapache/queue v1.1.0
go: downloading github.com/davecgh/go-spew v1.1.1
go: downloading github.com/rjeczalik/notify v0.9.2
go: downloading github.com/minio/simdjson-go v0.1.5
go: downloading github.com/pierrec/lz4 v2.4.0+incompatible
go: downloading github.com/eapache/go-xerial-snappy v0.0.0-20180814174437-776d5712da21
go: downloading github.com/golang/protobuf v1.4.0
go: downloading github.com/klauspost/pgzip v1.2.1
go: downloading github.com/lib/pq v1.7.0
go: downloading github.com/rcrowley/go-metrics v0.0.0-20181016184325-3113b8401b8a
go: downloading golang.org/x/text v0.3.3
go: downloading github.com/tidwall/match v1.0.1
go: downloading github.com/prometheus/client_model v0.0.0-20190812154241-14fe0d1b01d4
go: downloading go.uber.org/atomic v1.6.0
go: downloading google.golang.org/protobuf v1.22.0
go: downloading github.com/mailru/easyjson v0.7.1
go: downloading github.com/golang/snappy v0.0.1
go: downloading github.com/nats-io/nuid v1.0.1
go: downloading github.com/eapache/go-resiliency v1.1.0
go: downloading gopkg.in/ini.v1 v1.57.0
go: downloading github.com/eclipse/paho.mqtt.golang v1.2.0
go: downloading go.uber.org/multierr v1.5.0
go: downloading github.com/tidwall/sjson v1.0.4
go: downloading github.com/hashicorp/go-rootcerts v1.0.1
go: downloading github.com/willf/bitset v1.1.11
go: downloading github.com/bcicen/jstream v0.0.0-20190220045926-16c1f8af81c2
go: downloading github.com/hashicorp/go-uuid v1.0.1
go: downloading github.com/philhofer/fwd v1.0.0
go: downloading github.com/gomodule/redigo v2.0.0+incompatible
go: downloading github.com/coreos/go-systemd/v22 v22.0.0
go: downloading github.com/matttproud/golang_protobuf_extensions v1.0.1
go: downloading github.com/fatih/color v1.7.0
go: downloading github.com/elazarl/go-bindata-assetfs v1.0.0
go: downloading github.com/go-sql-driver/mysql v1.5.0
go: downloading github.com/minio/selfupdate v0.3.1
go: downloading github.com/coreos/go-semver v0.2.0
go: downloading gopkg.in/jcmturner/aescts.v1 v1.0.1
go: downloading github.com/nsqio/go-nsq v1.0.7
go: downloading github.com/beevik/ntp v0.2.0
go: downloading gopkg.in/jcmturner/dnsutils.v1 v1.0.1
go: downloading github.com/fatih/structs v1.1.0
go: downloading gopkg.in/yaml.v2 v2.2.8
go: downloading gopkg.in/jcmturner/rpc.v1 v1.1.0
go: downloading github.com/jcmturner/gofork v0.0.0-20190328161633-dc7c13fece03
go: downloading github.com/minio/sio v0.2.0
go: downloading github.com/xdg/scram v0.0.0-20180814205039-7eeb5667e42c
go: downloading gopkg.in/ldap.v3 v3.0.3
go: downloading github.com/ryanuber/go-glob v1.0.0
go: downloading github.com/json-iterator/go v1.1.10
go: downloading gopkg.in/jcmturner/goidentity.v3 v3.0.0
go: downloading github.com/xdg/stringprep v1.0.0
go: downloading github.com/nats-io/nats.go v1.10.0
go: downloading gopkg.in/asn1-ber.v1 v1.0.0-20181015200546-f715ec2f112d
go: downloading github.com/modern-go/reflect2 v1.0.1
go: downloading github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: downloading github.com/nats-io/jwt v0.3.2
go: downloading github.com/nats-io/nkeys v0.1.4
import cycle not allowed
package github.com/minio/minio
	imports github.com/minio/minio/cmd
	imports github.com/minio/minio/cmd/config
	imports github.com/minio/minio/cmd/logger
	imports github.com/minio/minio/cmd/config
make: *** [build] Error 1

```

--- a/cmd/config/config.go
+++ b/cmd/config/config.go
@@ -25,7 +25,6 @@ import (
 	"strings"
 
 	"github.com/minio/minio-go/v7/pkg/set"
-	"github.com/minio/minio/cmd/logger"
 	"github.com/minio/minio/pkg/auth"
 	"github.com/minio/minio/pkg/env"
 	"github.com/minio/minio/pkg/madmin"
@@ -451,7 +450,7 @@ func (c Config) Merge() Config {
 			}
 			if _, ok := cp[subSys]; !ok {
 				// A config subsystem was removed or server was downgraded.
-				logger.Info("config: removing unknown subsystem config %q", subSys)
+				fmt.Printf("config: removing unknown subsystem config %q\n", subSys)
 				continue
 			}
 			cp[subSys][tgt] = ckvs

--- a/cmd/config/config.go
+++ b/cmd/config/config.go
@@ -450,7 +450,7 @@ func (c Config) Merge() Config {
 			}
 			if _, ok := cp[subSys]; !ok {
 				// A config subsystem was removed or server was downgraded.
-				fmt.Printf("config: removing unknown subsystem config %q\n", subSys)
+				Logger.Info("config: ignoring unknown subsystem config %q\n", subSys)
 				continue
 			}
 			cp[subSys][tgt] = ckvs

--- a/cmd/config/logger.go
+++ b/cmd/config/logger.go
@@ -1,3 +1,19 @@
+/*
+ * MinIO Cloud Storage, (C) 2020 MinIO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
 package config
 
 import "context"

--- a/cmd/config/logger.go
+++ b/cmd/config/logger.go
@@ -1,0 +1,11 @@
+package config
+
+import "context"
+
+// Logger contains injected logger methods.
+var Logger = struct {
+	Info  func(msg string, data ...interface{})
+	LogIf func(ctx context.Context, err error, errKind ...interface{})
+}{
+	// Initialized via injection.
+}

--- a/cmd/logger/config.go
+++ b/cmd/logger/config.go
@@ -56,6 +56,12 @@ const (
 	EnvAuditWebhookAuthToken = "MINIO_AUDIT_WEBHOOK_AUTH_TOKEN"
 )
 
+// Inject into config package.
+func init() {
+	config.Logger.Info = Info
+	config.Logger.LogIf = LogIf
+}
+
 // Default KVS for loggerHTTP and loggerAuditHTTP
 var (
 	DefaultKVS = config.KVS{

